sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -488,6 +488,8 @@ class SparkSqlAstBuilder extends AstBuilder {
     } else {
       LocalTempView
     }
+    val qPlan: LogicalPlan = plan(ctx.query)
+    qPlan.setTagValue(LogicalPlan.NO_PARAM_ALLOWED_TAG, "CREATE VIEW body")
```

Comment:
since TreeNodeTag is not reliable. Shall we check whether `qPlan` contains parameter here?
```
    checkInvalidParameter(qPlan)
    qPlan.innerChildren.foreach {
      case p: LogicalPlan => checkInvalidParameter(p)
      case _ =>
    }
```
The method checkInvalidParameter is defined as
```
  private def checkInvalidParameter(plan: LogicalPlan): Unit = {
    plan.foreach { p =>
      if (p.expressions.exists(_.containsPattern(PARAMETER))) {
        p.failAnalysis(
          errorClass = "UNSUPPORTED_FEATURE.PARAMETER_MARKER_IN_UNEXPECTED_STATEMENT",
          messageParameters = Map("statement" -> "CREATE VIEW body"))
      }
    }
  }
```